src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
Thanks for all your efforts, Eric. The progress is great. We will fix the last details.

> The signature for the overridden ap() method isn't what you wrote above, but <B> Applicative<Validation<E, ?>, B> ap(Kind<Validation<E, ?>, ? extends Function<? super T, ? extends B>> f).

I can't reproduce it. When I copy-paste your code into IntelliJ IDEA (15.0.2 on a Mac with jdk 1.8.0_66) _and_ change the return type of the overridden `ap` method to `Validation<E, B>` it does compile fine (within IDE and with Maven on the command line), especially the `@Override` does also compile. (Note: `map` also may have a `@Override`)

Maybe our Applicative interfaces differ? Could you please check it?

```java
public interface Applicative<F extends Kind<F, ?>, T> extends Functor<T> {

    <U> Applicative<F, U> ap(Kind<F, ? extends Function<? super T, ? extends U>> f);

    @Override
    <U> Applicative<F, U> map(Function<? super T, ? extends U> f);
}
```

And Kind looks like this:

```java
package javaslang;

public interface Kind<TYPE extends Kind<TYPE, ?>, T> {
}
```

The Applicative interface would be a great addition to the algebra package. I questioned the whole package several times and tended to leave out these types because of several problems. It has taken many iterations but Monad does fit now very well into the library. If we do not give up too early we will achieve the same with Applicative.